### PR TITLE
Feature/add vue tests

### PR DESCRIPTION
**Pull request by OmgImAlexis**
This adds the needed setup helper for Ava and basic tests for the `app-link.vue` and `404.vue` components.

~Waiting on https://github.com/pymedusa/Medusa/pull/4526~

### CODE REVIEW

File: themes-default/slim/package.json [hunk 1]
```diff
@@ -23,14 +24,18 @@
       }
     }
   },
+  "dependencies": {},
   "devDependencies": {
     "autoprefixer": "^7.1.6",
-    "ava": "^0.23.0",
+    "ava": "0.24.0",
```
Comment (sharkykh) [inline themes-default/slim/package.json]: Caret removed by accident?

File: themes-default/slim/package.json [hunk 2]
```diff
@@ -113,18 +125,27 @@
...
     "files": [
-      "vue/test/**/*.js"
+      "test/specs/**/*.spec.js"
     ],
-    "source": [
-      "vue/src/**/*.{js,jsx}",
-      "!dist/**/*"
+    "sources": [
+      "views/**/*.vue",
+      "static/js/**/*.js"
```
Comment (sharkykh) [inline themes-default/slim/package.json]: Sources are incorrect, as the `views` folder doesn't/won't hold any `.vue` files according to #4374.
For now you can probably use the following, if that works...
```js
"sources": [
  "static/js/**/*.{js,vue}",
  "views/vue-components/*.mako"
]
```

File: themes-default/slim/test/specs/app-link.spec.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import test from 'ava';
+import Puex from 'puex';
+import { mount } from 'vue-test-utils';
+
+// Needs to be required otherwise nyc won't see it
+const AppLink = require('../../views/vue-components/app-link.vue');
```
Comment (sharkykh) [inline themes-default/slim/test/specs/app-link.spec.js]: It's not a `vue` file, it's a `mako` file.

File: themes-default/slim/test/specs/app-link.spec.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+test.beforeEach(t => {
+    t.context.base = 'http://localhost:8081/';
```
Comment (sharkykh) [inline themes-default/slim/test/specs/app-link.spec.js]: Please make the base configurable using environment variables or command line, falling back to `localhost:8081`.
This is a pain I have with the current `dredd` setup.

**Comment by sharkykh** [comment]
Are the snapshots the expected test results?

**Comment by OmgImAlexis** [comment]
> Are the snapshots the expected test results?

Yes, the snapshots are the results of the `t.snapshot(wrapper.html());`

I'd like all component tests to have a snapshot that shows the whole component at the end of the test so we know what's expected.

**Comment by OmgImAlexis** [comment]
Stupid autocorrect... and you can't edit review comments. 🙄 ugh.

File: themes-default/slim/views/schedule.mako [hunk 5]
```diff
@@ -10,6 +10,7 @@
 window.app = {};
 const startVue = () => {
     window.app = new Vue({
+        store,
         el: '#vue-wrap',
         store,
         router,
```
Comment (sharkykh) [inline themes-default/slim/views/schedule.mako]: Duplicate `store`.

File: themes/dark/templates/displayShow.mako [hunk 6]
```diff
@@ -19,6 +19,7 @@
 window.app = {};
 const startVue = () => {
     window.app = new Vue({
+        store,
         el: '#vue-wrap',
         store,
```
Comment (sharkykh) [inline themes/dark/templates/displayShow.mako]: Duplicate `store`.

**Comment by sharkykh** [comment]
Pages missing store:
- 400.mako - I know we're replacing it, but it could be chaos if we're not consistent
- 500.mako - Same as above.
- markdown.mako

**Pull request merged**